src/app/services/graph-client/MsalService.ts

```diff
@@ -0,0 +1,90 @@
+import { AuthenticationParameters } from 'msal/lib-commonjs/AuthenticationParameters';
+import { DEFAULT_USER_SCOPES } from '../graph-constants';
+import { msalApplication } from './MsalAgent';
+
+const loginType = getLoginType();
+const defaultUserScopes = DEFAULT_USER_SCOPES.split(' ');
+
+export async function logIn(): Promise<any> {
+  const loginRequest: AuthenticationParameters = {
+    scopes: defaultUserScopes,
+  };
+
+  if (loginType === 'POPUP') {
+    try {
+      await msalApplication.loginPopup(loginRequest);
+      const response = await msalApplication.acquireTokenPopup(loginRequest);
+      return response.accessToken;
+    } catch (error) {
+      return null;
+    }
+  } else if (loginType === 'REDIRECT') {
+    await msalApplication.acquireTokenRedirect(loginRequest);
+  }
+}
+
+export function logOut() {
+  msalApplication.logout();
+}
+
+/**
+ * Generates a new access token from passed in scopes
+ * @param {string[]} scopes passed to generate token
+ *  @returns {Promise.<any>}
+ */
+export async function acquireNewAccessToken(scopes: string[] = []): Promise<any> {
+  const hasScopes = (scopes.length > 0);
+  let listOfScopes = defaultUserScopes;
+  if (hasScopes) {
+    listOfScopes = scopes;
+  }
+
+  const loginRequest: AuthenticationParameters = {
+    scopes: listOfScopes,
+  };
+
+  try {
+    const response = await msalApplication.acquireTokenSilent(loginRequest);
+    return response;
+  } catch (error) {
+    if (requiresInteraction(error)) {
+      if (loginType === 'POPUP') {
+        try {
+          return msalApplication.acquireTokenPopup(loginRequest);
```

Comment:
It looks like the try catch here may be moot as the promise will not be awaited therefore be returned before an error can be thrown.
Maybe you want to do this?  `return await msalApplication.acquireTokenPopup(loginRequest);`
Checkout the SO question [here](https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise).

Reply:
👍